traits/ctraits.c

```diff
@@ -3064,6 +3064,22 @@ _trait_set_default_value(trait_object *trait, PyObject *args)
         return NULL;
```

Comment:
this is tangential but while looking up the difference between `PyErr_Format` and `PyErr_SetString`, I came across the fact that `PyErr_Format` returns `NULL` - so we don't need to explicitly return `NULL` here. Right?

Ref https://docs.python.org/3/c-api/exceptions.html#c.PyErr_Format

Reply:
Yes, we could! Not in this PR, but I'd be happy to look at a PR that made this and similar cleanups across `ctraits.c`.